web/gds-user-ui/src/components/ChooseAnOrganization/index.tsx

```diff
@@ -7,18 +16,73 @@ import { userSelector } from 'modules/auth/login/user.slice';
 import { useSelector } from 'react-redux';
 import { GrClose } from 'react-icons/gr';
 import { useNavigate } from 'react-router-dom';
-import React from 'react';
+import React, { useRef, useState, useEffect } from 'react';
+import Loader from 'components/Loader';
 
 // import { TransparentBackground } from 'components/TransparentBackground';
 function ChooseAnOrganization() {
-  const { organizations } = useOrganizationListQuery();
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  // const [prevPage, setPrevPage] = useState(0);
+  // const [orgList, setOrgList] = useState<any>([]);
+  const [wasLastPage, setWasLastPage] = useState<boolean>(false);
+  const { organizations, getAllOrganizations, isFetching } = useOrganizationListQuery(currentPage);
+  const { count, page_size, page } = organizations || {};
+  const listInnerRef = useRef<any>();
   const { user } = useSelector(userSelector);
   const navigate = useNavigate();
   const handleBack = (e: React.MouseEvent<HTMLElement>) => {
     e.preventDefault();
     navigate(-1);
   };
 
+  const NextPage = () => {
+    setCurrentPage(currentPage + 1);
+  };
+
+  const PreviousPage = () => {
+    setCurrentPage(currentPage - 1);
+  };
+
+  useEffect(() => {
+    if (currentPage !== 1) {
+      getAllOrganizations();
+    }
+  }, [currentPage, getAllOrganizations]);
+
+  useEffect(() => {
+    if (page && page_size && count && page * page_size >= count) {
+      setWasLastPage(true);
+    }
+    return () => {
+      setWasLastPage(false);
+    };
+  }, [page, page_size, count]);
```

Comment:
Put all this logic inside a hooks will make those logics more testable if you planned to write unit tests